testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java

```diff
@@ -22,64 +22,24 @@
 import io.stargate.it.storage.ExternalStorage;
 import io.stargate.starter.Starter;
 import java.io.IOException;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.net.InetAddress;
 import java.net.ServerSocket;
 import java.net.UnknownHostException;
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
-import net.bytebuddy.ByteBuddy;
-import net.bytebuddy.agent.ByteBuddyAgent;
-import net.bytebuddy.agent.builder.AgentBuilder;
-import net.bytebuddy.description.modifier.TypeManifestation;
-import net.bytebuddy.description.modifier.Visibility;
-import net.bytebuddy.implementation.MethodDelegation;
-import net.bytebuddy.implementation.bind.annotation.AllArguments;
-import net.bytebuddy.implementation.bind.annotation.Origin;
-import net.bytebuddy.implementation.bind.annotation.RuntimeType;
-import net.bytebuddy.matcher.ElementMatchers;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.objenesis.Objenesis;
-import org.objenesis.ObjenesisStd;
 import org.osgi.framework.BundleException;
-import org.osgi.framework.InvalidSyntaxException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/**
- * This class does some magic to allow access to OSGi loaded services for testing. The common
- * assumption is the classes loaded for testing in the local classpath are 1:1 with the classes
- * loaded in the services.
- *
- * <p>Under that assumption when a service is loaded it is "proxied"
- *
- * <p>ClassLoader 1 | Class A < -- Proxy -- > ClassLoader 2 | Class A
- *
- * <p>The unit test calls getOsgiService() which creates a proxy of the instance. Then when a call
- * is made to the proxy it intercepts the call and converts the passed arguments into a proxy for
- * OSGI's ClassPath version of the arguments and executes the call under the actual instance. The
- * result is then proxied back to the unit tests classpath version of the result class.
- *
- * <p>Special care is taken to handle collections, futures and enumerations.
- */
+/** This class manages starting Stargate OSGi containers. */
 @ExtendWith(ExternalStorage.class)
 @ClusterSpec(scope = SHARED)
 public class BaseOsgiIntegrationTest {
```

Comment:
since this is no longer responsible for accessing OSGi proxies, maybe we could rename this class? (to i.e. `BaseStargateIntegrationTest` or smth similar)

Reply:
I'm planning to convert it to a JUnit 5 extention (like `ExternalStorage`) is a follow-up PR... It will naturally be renamed there.

Reply:
👍 
`CqlSession` management could similarly be turned into an extension. It would be nice if each test could somehow control the scope of those extensions: per method or per class. I think that's possible if we use [programmatic registration](https://junit.org/junit5/docs/current/user-guide/#extensions-registration-programmatic).